pkg/flare/archive_nix.go

```diff
@@ -52,12 +52,9 @@ func (p permissionsInfos) statFiles() error {
 			return fmt.Errorf("while getting info of %s: %s", filePath, err)
 		}
 
-		sys, ok := fi.Sys().(*syscall.Stat_t)
-		if !ok {
-			// not enough information to append for this file
-			// might rarely happen on system not supporting this feature, but as
-			// we're building with !windows tag, shouldn't happen except for plan9
-			return fmt.Errorf("can't retrieve file uid/gid infos")
+		var sys syscall.Stat_t
+		if err := syscall.Stat(filePath, &sys); err != nil {
```

Comment:
I guess we don't really care but what happens now on plan9? Otherwise, it's much cleaner now!

Reply:
Where many other platforms returns a `*Stat_t` (Linux: https://golang.org/src/syscall/syscall_linux_amd64.go#L65 Darwin https://golang.org/src/syscall/zsyscall_darwin_amd64.go#L1186 ...), it looks like Plan9 is basically manipulating a slice of bytes without type: https://golang.org/src/syscall/syscall_plan9.go#L345
Thus I guess the agent won't compile on Plan9.